src/toxcore/dht/server.rs

```diff
@@ -407,6 +411,31 @@ impl Server {
             }
         }
     }
+    /**
+    handle received LanDiscovery packet, then create NodesRequest packet
+    and send back it to the peer.
+    */
+    fn handle_lan_discovery(&self, mut client: Client, packet: LanDiscovery) -> IoFuture<()> {
+        // TODO: get real friend pk
+        let friend_pk = PublicKey([15, 107, 126, 130, 81, 55, 154, 157,
+                                192, 117, 0, 225, 119, 43, 48, 117,
+                                84, 109, 112, 57, 243, 216, 4, 171,
+                                185, 111, 33, 146, 221, 31, 77, 118]);
+        
+        let mut state = self.state.write().expect("Failed to obtain ServerState lock");
+        let result = client.send_nodes_request(friend_pk);
+        state.peers_cache.insert(packet.pk, client);
+        result
+    }
+    /**
+    send LanDiscovery packet to all broadcast addresses
+    */
+    pub fn send_lan_discovery(&self) -> IoFuture<()> {
+        // addr is dummy, actual lan discovery packet sending logic exists in client
+        let addr: SocketAddr = "127.0.0.1:33445".parse().unwrap();
```

Comment:
Yet another reason why we should move tx and send_* functions from client to the server :)

Reply:
What do you mean?

Reply:
Ah
I got your saying.
Well, I think your comments deeply

Reply:
It requires a global refactoring. For now it is OK to create a dummy client, but in a week we have to rewrite server to never meet the code like:

```
// addr is dummy, actual lan discovery packet sending logic exists in client
let addr: SocketAddr = "127.0.0.1:33445".parse().unwrap();
```

Reply:
I mean that it's redundantly to store rx in each client because we have only one channel for udp connection. And if we remove tx from client and move send_* functions you won't have to create dummy address here. But it's not necessary for this pr.